### DISCUSSION TITLE
How to properly install packages

### DISCUSSION THREAD

**Discussion by rnnyrk** [vercel/turborepo#5888] [Help] [answered]
Hi there,

I'm a bit confused by the installation of packages. I have a setup similar to this:

- apps
- apps/docs
- apps/webapp
- packages/tailwind
- packages/components

All `package.json` files inside these folders include `autoprefixer`, `postcss`, `tailwind`, etc. As well as some other "duplicate packages". This can result in having different versions per folder. But I am not sure if I'm doing this right? I couldn't find this anywhere (or my google search is just off), but is it required to install the same packages in every folder and include the same dependency in multiple `package.json` files? Or can I just install a dependency in one folder and it's instantly referenced correctly throughout the whole turborepo?

**Comment by Ghirigoro** [comment, marked as answer]
My understanding is that package management is outside of Turbo's scope. In your project you would use npm/pnpm/yarn to manage packages and each of those would determine where packages are installed, how duplication is handled, etc. Turbo then uses whatever package manager is in your project to look up packages.

**Comment by mehulkar** [comment]
> but is it required to install the same packages in every folder

Turborepo does not require this, but we do recommend installing dependencies where they are used (and if that happens to be in every package, so be it)

> can I just install a dependency in one folder and it's instantly referenced correctly throughout the whole turborepo?

Turborepo doesn't help you load or use your dependencies, but from experience I can tell you that this will likely not work.
